Sharpmake/PackageReferences.cs

```diff
@@ -158,14 +158,14 @@ public enum AssetsDependency : uint
             None = 0,
             Compile = 1 << 0,
             Runtime = 1 << 1,
-            ContentFile = 1 << 2,
+            ContentFiles = 1 << 2,
```

Comment:
This will cause an API break, could you add back the old value, and mark it [Obsolete] ?

Reply:
I guess the new value should actually have the same flag, right?
```
            ContentFile = 1 << 2,
            ContentFiles = 1 << 2,
```

Reply:
It depends on how it's used in the code, if the value is store as int and converted back to the enum it could give us the wrong one. What I usually do when I deprecate things in flags like that is set the obsolete value to -1

Reply:
```
                foreach (var value in (AssetsDependency[])Enum.GetValues(typeof(AssetsDependency)))
                {
                    if (!dependency.HasFlag(value) || value == AssetsDependency.All || value == AssetsDependency.None)
                    {
                        continue;
                    }

                    var name = Enum.GetName(typeof(AssetsDependency), value);
                    // The first letter of assets values are in lower case (example: 'contentFiles')
                    yield return $"{char.ToLower(name[0])}{name.Substring(1)}";
                }
```
Its a bit problematic to have the old obsolete value in the enumeration as **GetFormatedAssetsDependency()** goes over all of them and picks a name by value, so I'm not sure what the best option is. Keeping old **ContentFile** we might need to add it into the if statement condition above to skip it it.

Reply:
Hum, why don't you sthg like this:
```csharp
public enum AssetsDependency : int // note the underlying type change
{
    ... // current values

    [Obsolete("Use " + nameof(ContentFiles) + " instead")]
    ContentFile = -1,
}
```

and in the loop that tests the values account for this one:
```csharp
if (/* current tests */ || (int)value == -1)
{
    continue;
}
```

Reply:
I'm not sure which is better, with you proposal, the user has to switch of warning as errors to be able to even get the value of -1 into that function, which might mean that they still want to keep using the old enum on purpose, but then the 
```c#
if (/* current tests */ || (int)value == -1)
{
    continue;
}
```
silently ignores the value and its name is not put into Assets.

Reply:
Yeah indeed, but your change also adds additional maintenance cost to the method translating the enum values to string.

Reply:
I'd be fine leaving it like this, as long as you add a check that all enum values were covered, to be notified in case we add new ones :)

Reply:
I actually wanted to write a unit test but the **GetFormatedAssetsDependency** function is marked **internal**, so I didnt. So where would I add such a check?

Reply:
Internals should be accessible from Sharpmake.UnitTests thanks to: https://github.com/ubisoft/Sharpmake/blob/eab7d8b2a026722cf6056cddc73be6fc528ca12e/Sharpmake/Properties/AssemblyInfo.cs#L54

Reply:
Added unit test, it revealed that also AssetsDependency.Analy**s**ers is "misspelled" according to the specs it should be "analy**z**ers". Not sure if that is actually a problem for MSBUILD or if it accepts both US or GB spelling.